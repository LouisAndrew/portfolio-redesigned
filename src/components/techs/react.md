---
templateKey: techs
title: React
name: ReactJS
icon: assets/react-1-logo-png-transparent.png
---
