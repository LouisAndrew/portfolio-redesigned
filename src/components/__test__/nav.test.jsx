import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import { ThemeProvider } from 'styled-components'

import Nav from '../nav'

describe('Navigation component on layout', () => {

        afterEach( cleanup )
        const theme = { center: () => { } }

        const renderWithStyles = ui => <ThemeProvider theme={theme}>{ui}</ThemeProvider>

        it('renders without crashing', () => {

                const div = document.createElement('div')
                ReactDOM.render( renderWithStyles(<Nav />), div )
        })

        it('renders correctly', () => {

                const { getByTestId } = render()
        })

        it('matches snapshot', () => {

                const tree = renderer.create().toJSON()
                expect(tree).toMatchSnapshot()
        })

})