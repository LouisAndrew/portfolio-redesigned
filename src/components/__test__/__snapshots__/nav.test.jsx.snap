// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component on layout matches snapshot 1`] = `null`;
